### PR TITLE
individual projects page

### PR DESCRIPTION
**Pull request by OnaMosimege**
Add individual project page, with styling and view

### CODE REVIEW

File: app/templates/app/project_detail.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          <a href="{% url 'institution_detail' project.institution.id %}">
+              {{ project.institution.name }}
+          </a>
+      </p>
+      <div>
+          <div class=project-date-row>
+            <div class=project-date-left-col>
+              <p>
+                <span class="icon-text"><i class="project-icon bi-calendar3"></i></span>
+                <span class="icon-text">Start Date: {{ project.start_date|date:"Y-m-d" }}<span class="icon-text">
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:25]: Please mark the string for translation.

File: app/templates/app/project_detail.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              </p>
+            </div>
+            <div class=project-date-right-col>
+              <p>
+                <span class="icon-text"><i class="project-icon bi-calendar3"></i></span>
+                <span class="icon-text">End Date: {{ project.end_date|date:"Y-m-d" }}<span class="icon-text">
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:31]: Please mark the string for translation.

File: app/templates/app/project_detail.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    </div>
+
+  </div>
+
+
+    <div class="detail-div-row">
+
+      <div class="col-sm-6 detail-col">
+        <div>
+            <h2>Subjects</h2>
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:52]: Please mark the string for translation.

File: app/templates/app/project_detail.html [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                  <span class="icon-text"><a href="{% url 'subject_detail' subject.id %}">{{ subject.name }}</a></span>
+                </li>
+                {% endfor %}
+            </ul>
+        </div>
+      </div>
+
+      <div class="col-sm-6 detail-col">
+        <div>
+          <h2>Languages</h2>
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:66]: Please mark the string for translation.

File: app/templates/app/project_detail.html [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    <div class="detail-div-row">
+
+      <div class="col-sm-6 detail-col">
+        <div>
+            <h2>Subjects</h2>
+            <ul>
+                {% for subject in project.subjects.all %}
+                <li>
+                  <span class="icon-text"><i class="project-icon bi-book"></i></span>
+                  <span class="icon-text"><a href="{% url 'subject_detail' subject.id %}">{{ subject.name }}</a></span>
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:57]: I believe this causes an n+1 queries. Please check with assertNumQueries in a test for this page.

File: app/templates/app/project_detail.html [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      </div>
+
+      <div class="col-sm-6 detail-col">
+        <div>
+          <h2>Languages</h2>
+          <ul>
+              {% for language in project.languages.all %}
+              <li>
+                <span class="icon-text"><i class="project-icon bi-vector-pen"></i></span>
+                <span class="icon-text"><a href="{% url 'language_detail' language.id %}">{{ language.name }}</a></span>
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:71]: I believe this causes an n+1 queries. Please check with assertNumQueries in a test for this page.

File: app/templates/app/project_detail.html [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+<div class="project-detail">
+  <p><a href="{% url 'projects' %}">{% trans "Projects" %}</a> > {{ project.name }}</p>
+
+  <div class=project-div-row>
+
+    <div class="col-sm-6 project-detail-left-col">
+      <h1>{{ project.name }}</h1>
+      <a href="{{ project.url }}">{{ project.url }}</a>
+      <p>
+          <a href="{% url 'institution_detail' project.institution.id %}">
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:16]: I believe this causes an n+1 queries. Please check with assertNumQueries in a test for this page.

File: app/templates/app/project_detail.html [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                <span class="icon-text">End Date: {{ project.end_date|date:"Y-m-d" }}<span class="icon-text">
+              </p>
+            </div>
+          </div>
+      </div>
+    </div>
+
+    <div class="col-sm-6 project-detail-right-col">
+      {% if project.logo %}
+      <img src="{{ project.logo.url }}" alt="{{ project.name }} logo"
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:40]: Please mark the alt text for translation.

File: app/templates/app/project_detail.html [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+           class="project-detail-logo">
+      {% endif %}
+    </div>
+
+  </div>
+
+
+    <div class="detail-div-row">
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:48]: Please review the indenting. This div starts further to the right than the one above closes. I'm trying to understand the relationship between `detail-div-row`, `project-div-row` and `project-detail`, and I'm hoping clear indentation might help.

**Review by friedelwolff — CHANGES_REQUESTED** [review]
Thanks. I'm ignoring the issues in the unfinished _detail() views, and I didn't test any of this at all.

File: app/templates/app/project_detail.html (hunk 1 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:25]: thanks, will do

File: app/templates/app/project_detail.html (hunk 3 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:52]: thanks, will do

File: app/templates/app/project_detail.html (hunk 4 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:66]: thanks, will do

File: app/templates/app/project_detail.html (hunk 8 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:40]: will do

File: app/templates/app/project_detail.html (hunk 9 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:48]: okay, will fix

File: app/templates/app/project_detail.html (hunk 5 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:57]: okay, made changes for this

File: app/templates/app/project_detail.html (hunk 6 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:71]: made changes for this

File: app/templates/app/project_detail.html (hunk 7 above)
Comment (OnaMosimege) [reply app/templates/app/project_detail.html:16]: will fix

File: app/templates/app/project_detail.html [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    <div class="col-sm-6 project-detail-left-col">
+      <h1>{{ project.name }}</h1>
+      <a href="{{ project.url }}">{{ project.url }}</a>
+      <p>
+          <a href="{% url 'institution_detail' project.institution.id %}">
+              {{ project.institution.name }}
+          </a>
+      </p>
+
+      <div>
```
Comment (friedelwolff) [inline app/templates/app/project_detail.html:21]: Just want to check if this `div` is required?

**Review by friedelwolff — APPROVED** [review]

**Pull request merged**